.github/workflows/build.yml

```diff
@@ -87,7 +87,7 @@ jobs:
     runs-on: ubuntu-latest
     environment:
       name: ${{ (github.ref == 'refs/heads/master' && 'production') || 'staging' }}
-    needs: [lint, test]
+    needs: [lint, test, build]
```

Comment:
What scenario does this prevent against? If the `build` job would fail, wouldn't the `yarn run build` step in this deployment job fail also and abort the deployment anyway?

Reply:
Thinking through this, I can go either way.

I will say our normal process is to build, and then once it's successful, we deploy. MPDX works this way.

Since we build on the CI when asking Amplify to deploy. Then Amplify runs a build to deploy.